receiver/prometheusremotewritereceiver/receiver_test.go

```diff
@@ -1819,3 +1782,104 @@ func buildMetaDataMapByID(ms pmetric.Metrics) map[string]map[string]any {
 	}
 	return result
 }
+
+// TestConcurrentRequestsforSameResourceAttributes reproduces the concurrency bug where subsequent requests
+// fail to reach the next consumer after the first successful request.
```

Comment:
```suggestion
// TestConcurrentRequestsforSameResourceAttributes asserts the receiver and its cache work even with concurrent requests
```

In a few weeks/months we may not remember that this bug ever existed and the comment might not make sense to future readers 😅